src/main/haskell/kore/src/Kore/Step/Simplification/Data.hs

```diff
@@ -10,51 +10,62 @@ Portability : portable
 module Kore.Step.Simplification.Data
     ( Simplifier
     , runSimplifier
+    , evalSimplifierWithTimeout
     , evalSimplifier
+    , defaultSMTTimeOut
     , PureMLPatternSimplifier (..)
     , CommonPureMLPatternSimplifier
     , SimplificationProof (..)
     ) where
 
+import Control.Monad.Reader
 import Kore.AST.Common
        ( PureMLPattern, Variable )
 import Kore.Step.OrOfExpandedPattern
        ( OrOfExpandedPattern )
 import Kore.Variables.Fresh
-
+import Kore.SMT.Config
 {-| 'SimplificationProof' is a placeholder for proofs showing that the
 simplification of a MetaMLPattern was correct.
 -}
 data SimplificationProof level = SimplificationProof
     deriving (Show, Eq)
 
-type Simplifier = Counter
+type Simplifier = ReaderT SMTTimeOut Counter
 
 {- | Run a simplifier computation.
 
   The result is returned along with the final 'Counter'.
 
  -}
 runSimplifier
-    :: Simplifier a
+    :: SMTTimeOut 
```

Comment:
Most editors can automatically remove whitespace at the end of line, please use that.